test/helpers/deep-extend.spec.js

```diff
@@ -39,3 +39,21 @@ describe("deepExtend", () => {
 		});
 	});
 });
+
+describe("Test merge options", () => {
+	it("should consider overrideDist option", () => {
+		const result = deepExtend({
+			b: 5,
+			c: 6
+		}, {
+			b: 10,
+			e: "Hello"
+		}, { overrideDest: false });
+
+		expect(result).toEqual({
```

Comment:
Please check that it really doesn't modify the source object, as well.

Reply:
That's exactly why it was.
because rule default settings must not override user schema property that already exist. it should only add new property/key to object - not overriding

Reply:
and `Object.assign({}, ....)` doesn't work here. because it creates a new Object every time and disrupt the cacher.

Reply:
I mean, checks the logic that it doesn't modify the original object, like: 

```js
	it("should consider overrideDist option", () => {
		const obj = {
			b: 5,
			c: 6
		};
		const result = deepExtend(obj, {
			b: 10,
			e: "Hello"
		}, { overrideDest: false });

		expect(result).toEqual({
			b: 5,
			c: 6,
			e: "Hello"
		});
		expect(obj).toEqual({
			b: 5,
			c: 6
		});        
	});
```

Reply:
It's ok, because `overrideDest` is false 
The key "b = 10" from source object is already exist in destination object and `deepExtand` function must not override this key. only add **new** key

Reply:
I understand what you said. My problem is that, if you didn't check that it doesn't touch the destination, and somebody changes this function in the future, we won't able to detect that this override logic still works well.

Reply:
By the way, just copy my previous code parts to the test & I'm fine to merge it :)

Reply:
Your previous test code will be fail. cause `deepExtend` function always return destination object as result.
This function always modifies destination (with any options). The only difference between options is the logic of extending.
I may have chosen a bad name for this option that you were confused about.

Reply:
Hmm, Ok, I will check it again.

Reply:
Ok, now I see what you tried to say :) I've renamed the option & changed the logic to the opposite.